addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -203,23 +286,34 @@ function ComposerFactory({ Entity }) {
         }
 
         /**
+         * @param {string} mentionKeyword
          * @private
-         * @param {string} content html content
-         * @returns {integer[]} list of mentioned partner Ids (not duplicate)
          */
-        _getMentionedPartnerIdsFromHtml(content) {
-            const parser = new window.DOMParser();
-            const node = parser.parseFromString(content, 'text/html');
-            const mentions = [...node.querySelectorAll('.o_mention')];
-            const allPartnerIds = mentions
-                .filter(mention =>
-                    (
-                        mention.dataset.oeModel === 'res.partner' &&
-                        !isNaN(Number(mention.dataset.oeId))
-                    )
-                )
-                .map(mention => Number(mention.dataset.oeId));
-            return [...new Set(allPartnerIds)];
+        async _getSuggestedPartners(mentionKeyword) {
+            const mentions = await this.env.rpc({
+                model: 'res.partner',
+                method: 'get_mention_suggestions',
+                kwargs: {
+                    limit: 3,
+                    search: mentionKeyword,
+                },
+            }, { shadow: true });
+
+            const suggestedPartners = mentions[0];
+            this.update({
+                suggestedPartners: [['insert-and-replace', suggestedPartners.map(data => this.env.entities.Partner.convertData(data))]],
+            });
```

Comment:
Just as reminder/double check, did we decide to ignore the fact that the order of the relation is not guaranteed in this case? cc @alexkuhn 
Note: I would prefer/suggest to merge the feature by ignoring this issue, and creating a new task for this problem if we consider it important.

Reply:
Let's ignore the ordering issue for the moment, and rely on implementation detail of replace that set order based on provided order of data/entities.